tests/bson/bson-value-var_export-001.phpt

```diff
@@ -0,0 +1,235 @@
+--TEST--
+MongoDB\BSON\Value: var_export tests
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/basic.inc';
+
+$tests = [
+    'null' => null,
+    'int32' => 2,
+    'int64' => new MongoDB\BSON\Int64(2),
+    'float' => 3.14,
+    'string' => 'foo',
+    'true' => true,
+    'false' => false,
+    'list' => [1, 2, 3],
+    'packedArray' => MongoDB\BSON\PackedArray::fromPHP([1, 2, 3]),
+    'struct' => ['foo' => 'bar'],
+    'object' => (object) ['foo' => 'bar'],
+    'document' => MongoDB\BSON\Document::fromPHP((object) ['foo' => 'bar']),
+    'binary' => new MongoDB\BSON\Binary('foo'),
+    'code' => new MongoDB\BSON\Javascript('foo'),
+    'codeWithScope' => new MongoDB\BSON\Javascript('foo', ['foo' => 'bar']),
+    'timestamp' => new MongoDB\BSON\Timestamp(1, 2),
+    'decimal128' => new MongoDB\BSON\Decimal128('1.2'),
+    'minKey' => new MongoDB\BSON\MinKey,
+    'maxKey' => new MongoDB\BSON\MaxKey,
+    'utcDateTime' => new MongoDB\BSON\UTCDateTime(1234567890),
+    'regex' => new MongoDB\BSON\Regex('foo'),
+    'dbPointer' => MongoDB\BSON\toPHP(MongoDB\BSON\fromJSON('{ "dbref": {"$dbPointer": {"$ref": "phongo.test", "$id" : { "$oid" : "5a2e78accd485d55b4050000" }  }} }'))->dbref,
+    'symbol' => MongoDB\BSON\toPHP(MongoDB\BSON\fromJSON('{ "symbol": {"$symbol": "test"} }'))->symbol,
+    'objectId' => new MongoDB\BSON\ObjectId('644a340b641850b1d0062d50'),
+    'undefined' => MongoDB\BSON\toPHP(MongoDB\BSON\fromJSON('{ "undefined": {"$undefined": true} }'))->undefined,
+];
+
+foreach ($tests as $name => $value) {
+    printf("Testing %s:\n", $name);
+
+    $bsonValue = MongoDB\BSON\Value::fromPHP($value);
+    echo var_export($bsonValue, true), "\n";
+}
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECTF--
+Testing null:
+%r\\?%rMongoDB\BSON\Value::__set_state(array(
```

Comment:
Is `Value::__set_state()` tested? We have tests for other `__set_state()` implementations.

I think various other Value methods might be missing tests as well, but I'm fine to put those off towards the end of the PR until all code changes are addressed.

Reply:
Tests for `__set_state` have been added. While there are no explicit test files for other methods, the main constructor tests also calls the `is<Type>` and `get<Type>` methods for each value that we're testing.